File: backend/internal/db/tables/recipe.go
package tables

import "github.com/google/uuid"

type Recipe struct {
	ID            uuid.UUID
	Name          string
	UniqueName    string
	Description   string
	OvenTemp      int
	EstimatedTime int
	Deleted       bool
	OwnedBy       uuid.UUID
	Portions      int
}

func (_ Recipe) StructName() string {
	return "Recipe"
}

File: frontend/src/api/Recipe.ts
import { Image } from "./Image";
import { Ingredient } from "./Ingredient";
import { Owner } from "./Owner";
import { Step } from "./Step";
import { Tag } from "./Tag";

export interface Recipe {
  id: string;
  name: string;
  uniqueName: string;
  description: string;
  ovenTemperature: number;
  estimatedTime: number;
  steps: Step[];
  ingredients: Ingredient[];
  images: Image[];
  author: Owner;
  tags: Tag[];
  portions: number;
}
